connectors/agent/protocol.py

```diff
@@ -60,20 +60,39 @@ async def apply_from_client(self):
         agent expects the components to respond within 30 seconds.
         See comment in https://github.com/elastic/elastic-agent-client/blob/main/elastic-agent-client.proto#L29
         """
-        logger.info("There's new information for the components/units!")
+        logger.info("New information available for components/units")
+
         if self.client.units:
             logger.debug("Client reported units")
+
             outputs = [
                 unit
                 for unit in self.client.units
                 if unit.unit_type == proto.UnitType.OUTPUT
             ]
-            if len(outputs) > 0 and outputs[0].config:
-                logger.debug("Outputs were found")
 
-                changed = self.agent_connectors_config_wrapper.try_update(outputs[0])
-                if changed:
-                    logger.info("Updating connector service manager config")
+            # Filter Elasticsearch outputs from the available outputs
+            elasticsearch_outputs = [
+                output
+                for output in outputs
+                if output.config and output.config.type == "elasticsearch"
```

Comment:
We can assume type exists on config object, see proto: https://github.com/elastic/elastic-agent-client/blob/86f524ddc40d8aae5a01244ae223fb87f8b33265/elastic-agent-client.proto#L172